pkg/util/httpgrpc/carrier.go

```diff
@@ -38,3 +45,28 @@ func GetParentSpanForRequest(tracer opentracing.Tracer, req *weaveworks_httpgrpc
 	}
 	return extracted, err
 }
+
+func GetParentSpanForQueryRequest(tracer opentracing.Tracer, req *queryrange.QueryRequest) (opentracing.SpanContext, error) {
+	if tracer == nil {
+		return nil, nil
+	}
+
+	carrier := opentracing.TextMapCarrier(req.Metadata)
+	extracted, err := tracer.Extract(opentracing.TextMap, carrier)
+	if err == opentracing.ErrSpanContextNotFound {
+		err = nil
```

Comment:
This feels... weird?
This will return a null SpanContext and a null error, right?

Reply:
That's how the original version was defined. See also [comment](https://github.com/grafana/loki/blob/c6ddd7044a7910addb2fb947cda43e9b3d5a070b/pkg/querier/worker/scheduler_processor.go#L146-L148) on main:

> Ignore errors here. If we cannot get parent span, we just don't create new one.

Actually, we should return an error here and let the caller decide whether to ignore it or not.

Reply:
Do you still want to do that, or leave it for a follow-up to land this?

Reply:
I'll do a follow up as it would change prod code that's not behind the feature flag.